app/routes/github_callback.rb

```diff
@@ -7,10 +7,12 @@ def self.registered(app)
           halt 400, "Must provide parameter 'code'" unless params[:code]
 
           begin
-            user = Authentication.perform(params[:code], github_client_id, github_client_secret, session[:target_profile])
+            user, access_token = Authentication.perform(params[:code], github_client_id, github_client_secret, session[:target_profile])
+            session[:access_token] = access_token
             login(user)
-          rescue StandardError => e
+          rescue Github::UnauthorizedUser => e
             Bugsnag.notify(e, nil, request)
+            session[:access_token] = nil
             flash[:error] = "We're having trouble with logins right now. Please come back later."
```

Comment:
Let's differentiate between the case where a user is unauthorized and the case where maybe there was a network connectivity error or something different. We probably still want to rescue `StandardError` or some other Faraday error that covers anything it might throw. In the case of a network error, this flash message and the Bugsnag notification make perfect sense. Only in a separate rescue block that rescues `Github::UnauthorizedUser` should we clear `session[:access_token]` and not report to Bugsnag.